bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/ReferenceResolverUtil.java

```diff
@@ -237,26 +232,29 @@ private static String resolvePattern(String reference, Map<String, ?> context) {
     }
 
     /**
-     * Found reference value should have meaningful String representation as it is part of configuration property which
-     * type is String
+     * Determines whether given Text is '${reference}'.
      *
-     * @param obj that is referenced in complex reference
-     * @return true if referenced value is valid complex reference type, false otherwise
+     * @param value to be evaluated
+     * @return True if this value is a '${reference}', false otherwise.
      */
-    private static boolean isSupportedPatternReferenceType(Object obj) {
-        return obj instanceof String || obj instanceof Number || obj instanceof Boolean;
-    }
-
-    private static boolean isReference(Object value) {
-        boolean result = false;
-        if (value instanceof String) {
-            String strVal = ((String) value).trim();
-            result = strVal.startsWith("$") && strVal.lastIndexOf("$") == 0 && strVal.length() > 1
-                    && strVal.charAt(1) != '{';
-        }
-        return result;
+    private static boolean isReference(String value) {
+        String trimmedVal = value == null ? null : value.trim();
+        return trimmedVal != null && trimmedVal.startsWith("${") && trimmedVal.lastIndexOf("${") == 0 // starts with
+                                                                                                      // '${' and it
+                                                                                                      // contains it
+                                                                                                      // only once
+                && trimmedVal.indexOf('}') == trimmedVal.lastIndexOf('}') // contains '}' only once
+                && trimmedVal.indexOf('}') == trimmedVal.length() - 1 && trimmedVal.length() > 3; // '}' is the last
+                                                                                                  // char and reference
+                                                                                                  // is not empty '${}'
     }
 
+    /**
+     * Determines whether given Text is '.....${reference}...'.
+     *
+     * @param value to be evaluated
+     * @return True if this value is a '.....${reference}...', false otherwise.
+     */
     private static boolean isPattern(String value) {
```

Comment:
the method does not seem to check for "isPattern", but rather for "containsPattern", right?